data_processing/services/radiology_service.py

```diff
@@ -0,0 +1,115 @@
+        "label_input_name": fields.String(description="Tag/name of label image record", required=True, example='user_segmentations'),
+        "enableAllImageTypes": fields.Boolean(description="Toggle all images", required=False, example=False),
+        "RadiomicsFeatureExtractor": fields.Raw(description="Key-value pairs for radiomics feature extractor", required=False, example={"interpolator": 'sitkBSpline'}),
+    }
+)
+
+# param models
+window_dicom_model = api.model("Scale and Window CT Dicom Datasets",
+    {
+        "job_tag": fields.String(description="Tag/name of output record", required=True, example='my_windowed_dicoms'),
+        "input_name": fields.String(description="Tag/name of input image record", required=True, example='dicoms'),
+        "window": fields.Boolean(description="Toggle window function", required=False, example=True),
+        "window.low_level": fields.Float(description="Low window value", required=False, example=-100),
+        "window.high_level": fields.Float(description="High window value", required=False, example=100),
+    }
+)
+
+# param models
+generate_scan_model = api.model("Generate Volumetric Image",
+    {
+        "job_tag": fields.String(description="Tag/name of output record", required=True, example='my_nrrd'),
+        "input_name": fields.String(description="Tag/name of input image record", required=True, example='dicoms'),
+        "file_ext": fields.String(description="A valid ITK image file extention", required=True, example='nrrd'),
+    }
+)
+
+@api.route('/mind/api/v1/window_dicom/<cohort_id>/<container_id>/submit', methods=['POST'])
+class API_window_dicom(Resource):
+    @api.expect(window_dicom_model, validate=True)
+    def post(self, cohort_id, container_id):
+        """Submit a scale and window CT dicom Job"""
+        job_id = str(uuid.uuid4())
+        future = executor.submit (window_dicom_with_container, cohort_id, container_id, request.json)
+        return f"Submitted job {job_id} with future {future}"
+
+@api.route('/mind/api/v1/extract_radiomics/<cohort_id>/<container_id>/submit', methods=['POST'])
+class API_extract_radiomics(Resource):
+    @api.expect(extract_radiomics_model, validate=True)
+    def post(self, cohort_id, container_id):
+        """Submit an extract radiomics Job"""
+        job_id = str(uuid.uuid4())
+        future = executor.submit (extract_radiomics_with_container, cohort_id, container_id, request.json)
+        return f"Submitted job {job_id} with future {future}"
+
+@api.route('/mind/api/v1/generate_scan/<cohort_id>/<container_id>/submit', methods=['POST'])
+class API_generate_scan(Resource):
+    @api.expect(generate_scan_model, validate=True)
+    def post(self, cohort_id, container_id):
+        """Submit a generate scan Job"""
+        job_id = str(uuid.uuid4())
+        future = executor.submit (generate_scan_with_container, cohort_id, container_id, request.json)
+        return f"Submitted job {job_id} with future {future}"
```

Comment:
It'd be good to standardize responses as a json with a status code and message/payload. 202 can be used as the status code for successful submission of async call.

Reply:
Ah, agreed, changed now.